### PR TITLE
Update dependencies, add .travis.yml

### LINKED ISSUE #10: Enable CI testing
We should run compilation and formatting checks.

### PR DESCRIPTION
**Pull request by ice1000**
.. and we need someone to actually setup this project on Travis CI.

Fixes #10

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      - cargo clippy --all -- -D clippy::all
+  # Since many users will use nightlies, also test that.
+  - rust: nightly
+
+script:
+  - cargo test --all -- --nocapture
+  # Validate benches still work.
+  - cargo bench --all -- --test
+  # Because failpoints inject failure in code path, which will affect all concurrently running tests, Hence they need to be synchronized, which make tests slow.
+  - cargo test --tests --features failpoint -- --nocapture
```
Comment (nrc) [inline .travis.yml]: I think we only need `cargo test` here

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+rust:
+
+matrix:
+  include:
+  # This build uses stable and checks rustfmt and clippy.
+  # We don't check them on nightly since their lints and formatting may differ.
+  - rust: stable
+    install:
+      - rustup component add rustfmt-preview
+      - rustup component add clippy-preview
```
Comment (nrc) [inline .travis.yml]: `-preview` suffixes are not necessary

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    before_script:
+      - cargo fmt --all -- --check
+      - cargo clippy --all -- -D clippy::all
```
Comment (nrc) [inline .travis.yml]: the `--all`s are not necessary, we should not check with `-D clippy::all`

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,39 @@
+branches:
+  only:
+    # This is where pull requests from "bors r+" are built.
+    - staging
+    # This is where pull requests from "bors try" are built.
+    - trying
+    # Enable building pull requests.
+    - master
```
Comment (nrc) [inline .travis.yml]: We don't have bors here, so I don't think these branches are useful

**Comment by nrc** [comment]
Thanks for adding this!

**Comment by ice1000** [comment]
Done

**Pull request merged**